r/vignettes/arrow.Rmd

```diff
@@ -72,6 +72,45 @@ to other applications and services that use Arrow. One example is Spark: the
 move data to and from Spark, yielding [significant performance
 gains](http://arrow.apache.org/blog/2019/01/25/r-spark-improvements/).
 
+# Object hierarchy
+
+## Metadata objects
```

Comment:
As I suggested before, I'd swap the order of data and metadata subsections, lead with a mapping of R vector/data.frame to Arrow versions, and then in the metadata section, invert the order of the table rows.

Reply:
Since this has been moved from the README to a less conspicuous place in the "Using the Arrow C++ Library in R" vignette, I don't think this is as important now. I also believe there are some good reasons to keep it in this order, chiefly the fact that it avoids forward references: reading from top to bottom, you never come across something that has not already been defined above. IMO we should keep this as is for now and plan to refine it for the next release.

Reply:
Re: forward references, I disagree that this is a virtue, I think it's more readable to start with the thing you care about and recurse down into the details as needed. But agreed that we don't have to resolve this today.